drivers/modem/gsm_ppp.c

```diff
@@ -121,6 +121,27 @@ NET_BUF_POOL_DEFINE(gsm_recv_pool, GSM_RECV_MAX_BUF, GSM_RECV_BUF_SIZE, 0, NULL)
 K_KERNEL_STACK_DEFINE(gsm_rx_stack, CONFIG_MODEM_GSM_RX_STACK_SIZE);
 K_KERNEL_STACK_DEFINE(gsm_workq_stack, CONFIG_MODEM_GSM_WORKQ_STACK_SIZE);
 
+#if DT_INST_NODE_HAS_PROP(0, mdm_restart_gpios)
+static const struct gpio_dt_spec restart_gpio =
+			GPIO_DT_SPEC_INST_GET(0, mdm_restart_gpios);
+
+#if !DT_INST_NODE_HAS_PROP(0, mdm_off_delay)
+#error "mdm-restart-gpios needs mdm-off-delay"
+#elif !DT_INST_NODE_HAS_PROP(0, mdm_on_delay)
+#error "mdm-restart-gpios needs mdm-on-delay"
+#endif
+
+#endif /* DT_INST_NODE_HAS_PROP(0, mdm-restart-gpios) */
+
+#if DT_INST_NODE_HAS_PROP(0, mdm_dtr_gpios)
+static const struct gpio_dt_spec dtr_gpio =
+			GPIO_DT_SPEC_INST_GET(0, mdm_dtr_gpios);
+#endif
+#if DT_INST_NODE_HAS_PROP(0, mdm_wdisable_gpios)
+static const struct gpio_dt_spec wdisable_gpio =
+			GPIO_DT_SPEC_INST_GET(0, mdm_wdisable_gpios);
+#endif
```

Comment:
please use driver's config instead (to align with how drivers are implemented in general)